osquery/core/watcher.cpp

```diff
@@ -527,6 +528,10 @@ void WatcherRunner::createWorker() {
   // Get the complete path of the osquery process binary.
   boost::system::error_code ec;
   auto exec_path = fs::system_complete(fs::path(qd[0]["path"]), ec);
+  if (!pathExists(exec_path.string()).ok()) {
```

Comment:
The signature of function `pathExists` as far as I know looks like: `Status pathExists(const fs::path& path)`. It means this is a double conversion from path to string and from string to path.  So, you can call it just: `pathExists(exec_path).ok()`.

Reply:
👍